sentry-android-core/src/test/java/io/sentry/android/core/ActivityFramesTrackerTest.kt

```diff
@@ -102,15 +110,166 @@ class ActivityFramesTrackerTest {
         assertEquals(slowFrames!!.value, 5f)
     }
 
+    @Test
+    fun `sets slow and frozen frames even if start was null`() {
+        val sut = fixture.getSut()
+        val arrayAll = SparseIntArray()
+        arrayAll.put(1, 100)
+        arrayAll.put(20, 5)
+        arrayAll.put(705, 6)
+        val array = arrayOf(arrayAll)
+
+        whenever(fixture.aggregator.metrics).thenReturn(null, array)
+
+        sut.addActivity(fixture.activity)
+        sut.setMetrics(fixture.activity, fixture.sentryId)
+
+        val metrics = sut.takeMetrics(fixture.sentryId)
+
+        val totalFrames = metrics!!["frames_total"]
+        assertEquals(totalFrames!!.value, 111f)
+
+        val frozenFrames = metrics["frames_frozen"]
+        assertEquals(frozenFrames!!.value, 6f)
+
+        val slowFrames = metrics["frames_slow"]
+        assertEquals(slowFrames!!.value, 5f)
+    }
+
+    @Test
+    fun `different activities have separate counts - even when called out of order`() {
+        val sut = fixture.getSut()
+        val array0 = SparseIntArray().also {
+            it.put(1, 100)
+            it.put(20, 5)
+            it.put(705, 6)
+        }.let { arrayOf(it) }
+        val array1 = SparseIntArray().also {
+            it.put(1, 110)
+            it.put(20, 6)
+            it.put(705, 7)
+        }.let { arrayOf(it) }
+        val array2 = SparseIntArray().also {
+            it.put(1, 115)
+            it.put(20, 8)
+            it.put(705, 9)
+        }.let { arrayOf(it) }
+        val array3 = SparseIntArray().also {
+            it.put(1, 135)
+            it.put(20, 11)
+            it.put(705, 12)
+        }.let { arrayOf(it) }
+
+        val otherActivity = mock<Activity>()
+        val otherSentryId = SentryId()
+
+        whenever(fixture.aggregator.metrics).thenReturn(array0, array1, array2, array3)
+
+        sut.addActivity(fixture.activity)
+        sut.addActivity(otherActivity)
+        sut.setMetrics(fixture.activity, fixture.sentryId)
+        sut.setMetrics(otherActivity, otherSentryId)
+
+        val metrics = sut.takeMetrics(fixture.sentryId)
+        val otherMetrics = sut.takeMetrics(otherSentryId)
```

Comment:
`l`: If we do this, we don't need them everywhere below.
```suggestion
        val metrics = sut.takeMetrics(fixture.sentryId)!!
        val otherMetrics = sut.takeMetrics(otherSentryId)!!
```